lib/HALTarget/IMU.cpp

```diff
@@ -0,0 +1,211 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  IMU implementation
+ * @author Juliane Kerpe <juliane.kerpe@web.de>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "IMU.h"
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+bool IMU::init()
+{
+    bool isInitSuccessful = m_imu.init();
+    if (isInitSuccessful) {
+        m_imu.enableDefault();
+    }
+    
+    // TD074    Make sure that a Full Scale of 245 dps of the gyros are enough
+    // TD075	Make sure that it is valid to make the Full Scale setting in IMU::init()
+    switch(m_imu.getType())
+    {
+    case Zumo32U4IMUType::LSM303D_L3GD20H:
+        // 0x00  means +/- 245 dps according to L3GD20H data sheet
+        m_imu.writeReg(L3GD20H_ADDR, L3GD20H_REG_CTRL4, 0x00);
+        break;
+    case Zumo32U4IMUType::LSM6DS33_LIS3MDL:
+        // 0x00  means +/- 245 dps according to LSM6DS33 data sheet
+        m_imu.writeReg(LSM6DS33_ADDR, LSM6DS33_REG_CTRL2_G, 0x00);
+        break;
+    default:
+        ;
+    }
+    return isInitSuccessful;
+}
+
+void IMU::enableDefault()
+{
+    m_imu.enableDefault();
+}
+
+void IMU::configureForTurnSensing()
+{
+    m_imu.configureForTurnSensing();
+}
+
+void IMU::readAcc()
+{
+    while(!m_imu.accDataReady()){}
+    m_imu.readAcc();
+    m_accelerometerValues.x = m_imu.a.x;
+    m_accelerometerValues.y = m_imu.a.y;
+    m_accelerometerValues.z = m_imu.a.z;
+}
+
+void IMU::readGyro()
+{
+    while(!m_imu.gyroDataReady()){}
```

Comment:
Please use multiple rows & add comment that nothing is to be done while waiting.

Reply:
Resolved since the line has been deleted and replaced by a non-blocking retrieval